glommio/src/sys/source.rs

```diff
@@ -225,3 +239,112 @@ impl Drop for Source {
         }
     }
 }
+
+pub struct SharedSource {
+    source: Source,
```

Comment:
I am surprised by this.
The `Source` already has a list of results and wakers, so we are duplicating all of this here.

Keep in mind that those structs are allocated, so we'd like them to be as small as possible.
I thought you were going to create this from its building blocks.

Reply:
> The Source already has a list of results and wakers

Not quite. `Source` keeps track of the result and **one** waker. We need a list of those. This is the primary reason that justifies the existence of `SharedSource`